modules/rostests/apitests/iphlpapi/icmp.c

```diff
@@ -82,9 +82,54 @@ test_IcmpCloseHandle(void)
     ok_err(ERROR_INVALID_HANDLE);
 }
 
+static
+void
+test_IcmpSendEcho(void)
+{
+    HANDLE hIcmp;
+    unsigned long ipaddr = INADDR_NONE;
+    DWORD bRet = 0;
+    char SendData[32] = "Data Buffer";
+    PVOID ReplyBuffer = NULL;
+    DWORD ReplySize = 0;
+
+    SetLastError(0xDEADBEEF);
+    hIcmp = IcmpCreateFile();
+    if (hIcmp == INVALID_HANDLE_VALUE)
+    {
+        skip("IcmpCreateFile failed unexpectedly: %lu\n", GetLastError());
+        return;
+    }
+
+    ipaddr = 0x08080808; // 8.8.8.8
+
+    ReplySize = sizeof(ICMP_ECHO_REPLY);
+    ReplyBuffer = malloc(ReplySize);
+    SetLastError(0xDEADBEEF);
+    bRet = IcmpSendEcho(hIcmp, ipaddr, SendData, sizeof(SendData), 
+        NULL, ReplyBuffer, ReplySize, 5000);
+
+    free(ReplyBuffer);
+    ok(!bRet, "IcmpSendEcho succeeded unexpectedly\n");
+    bRet = GetLastError();
```

Comment:
Variables aren't in short supply, you can use a new one for the error code ;)
(alternatively rename this one to something more generic like "result")

Reply:
Improved! :)